File: python/1.3.3_SteppingMotor.py
#!/usr/bin/env python3

import RPi.GPIO as GPIO
import time 

motorPins = (12, 16, 18, 22)    #pins connected to four phase
antiClk = (0x01,0x02,0x04,0x08) 
clk = (0x08,0x04,0x02,0x01)

def setup():
    GPIO.setmode(GPIO.BOARD)       # Numbers GPIOs by physical location
    for pin in motorPins:
        GPIO.setup(pin,GPIO.OUT)

  
def moveSteps(direction, steps):
    for step in range(steps):
        for j in range(0,4,1):  #cycle according to power supply order 
            for i in range(0,4,1):  #assign to each pin
                if (direction == 1):#clockwise
                    GPIO.output(motorPins[i],((antiClk[j] == 1<<i) and GPIO.HIGH or GPIO.LOW))
                else :              #anticlockwise
                    GPIO.output(motorPins[i],((clk[j] == 1<<i) and GPIO.HIGH or GPIO.LOW))
            time.sleep(0.003)    #the delay can not be less than 3ms

#function used to stop rotating
def motorStop():
    for i in range(0,4,1):
        GPIO.output(motorPins[i],GPIO.LOW)
            
def loop():
    while True:
        moveSteps(1,512)
        time.sleep(1)
        moveSteps(0,512) 
        time.sleep(1)

def destroy():
    GPIO.cleanup()             # Release resource

if __name__ == '__main__':     # Program start from here
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the program destroy() will be  executed.
        destroy()

File: c/1.3.3/1.3.3_SteppingMotor.c
/**********************************************************************
* Filename    : 1.3.3_SteppingMotor.c
* Description : 
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/
#include <stdio.h>
#include <wiringPi.h>

const int motorPins[]={1,4,5,6};    //pins connected to four phase
const int antiClk[]={0x01,0x02,0x04,0x08};  
const int clk[]={0x08,0x04,0x02,0x01};   

void moveSteps(int direction, int steps){
    int step;
    int i=0,j=0;
    for(step=0;step<steps;step++){
        for (j=0;j<4;j++){  //cycle according to power supply order 
        for (i=0;i<4;i++){  //assign to each pin
            if(direction == 1)    //clockwise
                digitalWrite(motorPins[i],antiClk[j] == (1<<i));
            else        //anticlockwise
                digitalWrite(motorPins[i],clk[j] == (1<<i));
        }
        delay(3); //the delay can not be less than 3ms
    }
    }
}
void motorStop(){   //stop rotating
    int i;
    for(i=0;i<4;i++){
        digitalWrite(motorPins[i],LOW);
    }   
}
int main(void){
    int i;
	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|             Stepping motor           |\n");
	printf("|    ------------------------------    |\n");
	printf("|          IN1 connect to # 18         |\n");
    printf("|          IN2 connect to # 23         |\n");
    printf("|          IN3 connect to # 24         |\n");
    printf("|          IN4 connect to # 25         |\n");
	printf("|                                      |\n");
	printf("|     Controlling a Stepping motor     |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");
    if(wiringPiSetup() == -1){ //when initialize wiring failed,print message to screen
        printf("setup wiringPi failed !");
        return 1; 
    }
    for(i=0;i<4;i++){
        pinMode(motorPins[i],OUTPUT);
    } 

    while(1){
        moveSteps(1,512); 
        delay(1000);
        moveSteps(0,512);
        delay(1000);
    }
    return 0;
}

